### PR TITLE
feat: Add random binary and octal number generation

### LINKED ISSUE #184: feat: add binary and octal random number generation
### Clear and concise description of the problem

`faker.random.[...]` already generates random number, bigInt, float, hexadecimal, etc. But it lacks octal and binary numbers. This can be added. 

### Suggested solution

This is an easy task. Either use the [`faker.datatype.hexadecimal()`](https://github.com/faker-js/faker/blob/4a69d6dc295153e037c15e084e55aef1e662096b/src/datatype.ts#L179) method as a reference or generate a random number using [`faker.datatype.number()`](https://github.com/faker-js/faker/blob/4a69d6dc295153e037c15e084e55aef1e662096b/src/datatype.ts#L27) and convert it into binary or octal.

A simple way to convert decimal to binary is:

```js
console.log(Number((128).toString(2))) // 10000000
```

Similarly decimal to octal conversion is:

```js
console.log(Number((128).toString(8))) // 200
```

(this is just one method to do so)

### Alternative

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mbokinala**
closes #184

### CODE REVIEW

File: src/datatype.ts [hunk 1]
```diff
@@ -208,6 +208,45 @@ export class Datatype {
...
   }
 
+  /**
+   * Returns random octal string of specified length.
+   *
+   * @method faker.datatype.octal
+   * @param count defaults to 1
+   */
+  octal(count: number = 1): string {
+    let wholeString = '';
```
Comment (poyea) [inline src/datatype.ts:218]:
```suggestion
    let octalString = '';
```

File: src/datatype.ts [hunk 2]
```diff
@@ -208,6 +208,45 @@ export class Datatype {
...
+  }
+
+  /**
+   * Returns random binary string of specified length.
+   *
+   * @method faker.datatype.binary
+   * @param count defaults to 1
+   */
+  binary(count: number = 1): string {
+    let wholeString = '';
```
Comment (poyea) [inline src/datatype.ts:242]:
```suggestion
    let binaryString = '';
```

File: test/datatype.unit.js [hunk 3]
```diff
@@ -232,6 +232,34 @@ describe('datatype.js', function () {
...
+    var binary = faker.datatype.binary;
+
+    it('generates a single binary charcter when no additional argument is provided', function () {
+      var binaryNum = binary();
+      assert.ok(binaryNum.match(/^(0b)[01]{1}$/i));
+    });
+
+    it('generates a random binary string', function () {
+      var hex = binary(12);
+      assert.ok(hex.match(/^(0b)[01]+$/i));
```
Comment (poyea) [inline test/datatype.unit.js:245]:
```suggestion
      assert.ok(hex.match(/^0b[01]{12}$/i));
```

File: test/datatype.unit.js [hunk 4]
```diff
@@ -232,6 +232,34 @@ describe('datatype.js', function () {
     });
   });
 
+  describe('binary', function () {
+    var binary = faker.datatype.binary;
+
+    it('generates a single binary charcter when no additional argument is provided', function () {
+      var binaryNum = binary();
+      assert.ok(binaryNum.match(/^(0b)[01]{1}$/i));
```
Comment (poyea) [inline test/datatype.unit.js:240]:
```suggestion
      assert.ok(binaryNum.match(/^0b[01]$/i));
```

File: src/datatype.ts [hunk 5]
```diff
@@ -208,6 +208,45 @@ export class Datatype {
     return '0x' + wholeString;
   }
 
+  /**
+   * Returns random octal string of specified length.
+   *
+   * @method faker.datatype.octal
+   * @param count defaults to 1
+   */
+  octal(count: number = 1): string {
```
Comment (poyea) [inline src/datatype.ts:217]: I'd suggest `length`, but it's OK for consistency with `hexaDecimal`.

File: docs/api/datatype.md [hunk 6]
```diff
@@ -123,3 +123,37 @@ faker.datatype.hexaDecimal(); // '0xA' (string)
...
 ```
+
+## Binary
+
+Generates random binary (base-2) number.
+
+::: tip
+| Param | Type | Default |
+| ----- | ----- | :---------------: |
+| length | number | 1 |
```
Comment (poyea) [inline docs/api/datatype.md:134]: and it's `length` here

File: src/datatype.ts (hunk 5 above)
Comment (mbokinala) [reply src/datatype.ts:217]: I agree, I'll update this

File: docs/api/datatype.md (hunk 6 above)
Comment (mbokinala) [reply docs/api/datatype.md:134]: I modeled the two new methods on the existing `hexaDecimal` method, but it looks like the `hexaDecimal` method uses the `count` parameter while its docs say `length`. I will update all three methods and docs to use `length`.

File: src/datatype.ts [hunk 7]
```diff
@@ -174,11 +174,11 @@ export class Datatype {
    * hexaDecimal
    *
    * @method faker.datatype.hexaDecimal
-   * @param count defaults to 1
+   * @param length defaults to 1
    */
-  hexaDecimal(count: number = 1): string {
+  hexaDecimal(length: number = 1): string {
     let wholeString = '';
```
Comment (poyea) [inline src/datatype.ts:180]: Maybe `hexaString` too? It's very minor though.

**Review by poyea — APPROVED** [review]
looks great!

File: src/datatype.ts (hunk 7 above)
Comment (mbokinala) [reply src/datatype.ts:180]: Ok, I'll edit it

**Comment by Shinigami92** [comment]
We're starting work on the `v6.2` milestone, but this PR needs a rebase.
Please rebase and resolve all conflicts. Until then, we're moving it to the `v6` milestone, which now serves as the "v6-Backlog".
If this PR no longer contains any conflicts and can certainly be a target for the next release, we will move it to the then active milestone `v6.x`.

**Comment by petrzjunior** [comment]
Could we also provide a version without the `0b`, `0o`, `0x` prefix? Possibly via optional parameter.
In my usecase, I need to repeatedly write `.substring(2)` to remove it.

```js
Buffer.from(faker.datatype.hexadecimal(20).substring(2), 'hex')
```

**Comment by Shinigami92** [comment]
> Could we also provide a version without the `0b`, `0o`, `0x` prefix? Possibly via optional parameter.
> 
> In my usecase, I need to repeatedly write `.substring(2)` to remove it.
> 
> 
> 
> ```js
> 
> Buffer.from(faker.datatype.hexadecimal(20).substring(2), 'hex')
> 
> ```

Don't have to much time yet to go into details and I have other prios right now, but yes I would like to not prefix it by default at all but have just the option to add the prefix on demand via an option argument. Consistent for all of these functions!

**Comment by xDivisionByZerox** [comment]
I will close this PR since it's heavily outdated. I suggest opening another PR that at least has the 'new' test strategy and directory structure implemented.

If anyone really wants to keep this alive, feel free to reopen it.

**Pull request closed without merging**